File: Objc in Swift app/Objective-C/DelegateInObjC/TaskHandler.m
//
//  TaskHandler.m
//  Objc in Swift app
//
//  Created by Cong Le on 2/1/24.
//


#import "TaskHandler.h"

@implementation TaskHandler
- (void)taskManager:(TaskManager *)taskManager didCompleteTaskWithResult:(NSString *)result {
    NSLog(@"From Objective-C: Delegate received task completion with result: %@", result);
}
@end

File: Objc in Swift app/DelegationInSwift/TaskHandler.swift
//
//  TaskHandler.swift
//  Objc in Swift app
//
//  Created by Cong Le on 2/1/24.
//

import Foundation

class TaskHandler: TaskDelegate {
    func didCompleteTask(_ sender: TaskManager, result: String) {
        print("From Swift: Delegate received task completion with result: \(result)")
    }
}
